### DISCUSSION TITLE
Fine Classing

### DISCUSSION THREAD

**Discussion by joejnt** [teamalgoritma/community#551] [Q&A] [answered]
Dear team Algoritma,

Saya melakukan proses fine classing dengna menjalankan syntax berikut
df_numeric <- df_cln %>%
  select_if(is.numeric) %>% 
  names()

fine_class <- woebin(df_cln, 
                    y = "PROB_DEFAULT", 
                    x = df_numeric, 
                    positive = 'Y', # the value in gb_flag that indicates default
                    method = "freq", # frequency method
                    bin_num_limit = 20)  # the max number of fine bins

<img width="361" alt="image" src="https://user-images.githubusercontent.com/103427989/220008351-93b36343-1c47-42e3-9dbb-13abc5fe215c.png">

Bagaimana cara saya bisa download hasil dari proses woebin() ke excel, saya sudah menggunakan syntax berikut
woebin_list <- woebin_plyr(df_cln, fine_class, type = "list")
write.xlsx(woebin_list, "woebin_output.xlsx")

hasilnya error.

Tambahan, saya mendapatkan permasalahan untuk menentukan variable mana yang penting yang akan digunakan sebagai pemodelan dimana var target adalah "0" debitur yang lancara dan "1" debitur yang tidak lancar. untuk melakukan fine classing saya membuat perhitungan WoE dengan syntax sebagai berikut:
df_numeric <- df_cln %>%
  select_if(is.numeric) %>% 
  names()
fine_class <- woebin(df_cln, 
                    y = "PROB_DEFAULT", 
                    x = df_numeric, 
                    positive = 1,
                    method = "freq",
                    bin_num_limit = 20)

dan nilai IV dengan syntax sebagai berikut:
iv <- map_df(fine_class, ~pluck(.x, 10, 1)) %>%
  pivot_longer(everything(), names_to = "variabel", values_to = "iv")

kemudian saya ingin menggabungkan hasil perhitungan WoE dengan IV dengan syntax sebagai berikut:
df_woe <- woebin_plot(fine_class) %>% woebin(df_cln$PROB_DEFAULT)

namun hasilnya error. 
<img width="216" alt="image" src="https://user-images.githubusercontent.com/103427989/220043959-d835e06f-1d7f-4229-8843-5d8061bf7dea.png">

Tujuan saya adalah untuk mencari variabel-variabel apa saja yang akan dipertahankan atau var mana yang akan dibuang sebelum pemodelan.

Juga bagaimana ya cara untuk membaca woebin_plot(), contoh seperti dibawah:
<img width="332" alt="image" src="https://user-images.githubusercontent.com/103427989/220080399-27a637e9-f3ac-4086-a39e-4ef11bc7c84e.png">

Mohon saran dan petunjuknya.

Best regards,
JT

**Comment by finesaaa** [comment]
Dear @joejnt,
Terima kasih atas pertanyaannya, akan kami coba carikan referensi terlebih dahulu ya pak

**Comment by ranydwi** [comment, marked as answer]
Dear Pak @joejnt 
Terima kasih untuk pertanyaan dan kesediannya untuk menunggu.

Saya ingin bertanya sebelumnya, error yang muncul ketika bapak ingin melakukan export proses `woebin()` ke excel seperti apa ya? Selanjutnya silahkan ditampilkan terlebih dahulu object `woebin_list` yang sudah bapak buat.

Error yang bapak alami ini disebabkan karena terdapat proses yang gagal dilakukan ketika melakukan exporting, karena list dalam object `woebin_list` masih belum benar.

Selanjutnya, referensi untuk proses plot saya dapat berikan referensi berikut Pak. https://rpubs.com/harendra_16/Credit_Scoring_modelling

Semoga membantu :)

**Reply by joejnt** [reply]
Dear ibu @ranydwi 

Untuk permasalahan proses export hasil woebin() sudah saya dapatkan. Terima kasih untuk referensi atas proses plotnya.

Sedikit tambahan pertanyaan, jika dalam pemodelan Log. Regresion hasil evaluasi yang saya dapatkan seperti ini 

Data test
<img width="215" alt="image" src="https://user-images.githubusercontent.com/103427989/220550526-78a963af-cf7c-4384-b171-188e3a4c56f4.png">

Data train
<img width="218" alt="image" src="https://user-images.githubusercontent.com/103427989/220550609-ee9ee8b7-350b-45ec-a1c2-9332b2ee786c.png">

Apakah ada cara untuk melakukan tuning model ?

Adapun yang saya lakukan adalah dengan cara sebagai berikut:
**Pertama:** untuk split data train saya sudah melakukan seleksi dengan menghilangkan tipe data selain numerik, hanya menyisakan 1 var character pada kol 1 yaitu nmor rekening yang saya anggap unique. dengan syntax berikut:
df_model <- dev_woe2 %>% 
  select(-c(BRANCH_NAME, BRANCH_CODE, CUSTOMER_NAME, CIF, DATE_BIRTH, PEKERJAAN, SUMBER_DANA, GENDER, PLAN_DESCRIPTION, 
            TANGGAL_BUKA, TGL_JTH_TEMPO, AUDIT_SAMPLE, KOLEKTABILITAS, KATEGORI_RISIKO, KATEGORI_WATCHLIST, RANGE_LATE_FREQ,
            RANGE_AGE))

**Kedua:** 
saya menjalankan syntax ini untuk melakukan tuning 
# define control parameters for feature selection
ctrl <- rfeControl(functions = rfFuncs,
                   method = "cv",
                   number = 10)

Namun hasilnya error pada saat menjalankan syntax ini 
# select features using recursive feature elimination (RFE)
rfe <- rfe(train[,-1], train$pro, sizes = c(1:ncol(trainData)-1), rfeControl = ctrl)

<img width="473" alt="image" src="https://user-images.githubusercontent.com/103427989/220551861-d9b07847-3fdf-4568-b667-d74b425600aa.png">

Mohon sarannya.

Best regards,
JT
